wave_utils/include/wave/utils/logging.hpp

```diff
@@ -1,22 +1,19 @@
-#ifndef __wave_UTILS_LOGGING_HPP__
-#define __wave_UTILS_LOGGING_HPP__
-
-#include <stdio.h>
-#include <iostream>
-#include <cstdarg>
-
+#ifndef __WAVE_UTILS_LOGGING_HPP__
+#define __WAVE_UTILS_LOGGING_HPP__
 
 namespace wave {
 
-#define ERROR_FORMAT "[ERROR] (%s:%d) "
-#define WARN_FORMAT "[WARN] (%s:%d) "
-#define INFO_FORMAT "[INFO] "
+#define __FILENAME__ \
+    (strrchr(__FILE__, '/') ? strrchr(__FILE__, '/') + 1 : __FILE__)
+
+#define log_err(M, ...)                \
+    fprintf(stderr,                    \
+            "[ERROR] [%s:%d] " M "\n", \
+            __FILENAME__,              \
+            __LINE__,                  \
+            ##__VA_ARGS__)
 
-#define LOG_ERROR(M, ...) \
-  fprintf(stderr, ERROR_FORMAT M "\n", __func__, __LINE__, ##__VA_ARGS__)
-#define LOG_WARN(M, ...) \
-  fprintf(stderr, WARN_FORMAT M "\n", __func__, __LINE__, ##__VA_ARGS__)
-#define LOG_INFO(M, ...) fprintf(stderr, INFO_FORMAT M "\n", ##__VA_ARGS__)
+#define log_info(M, ...) fprintf(stdout, "[INFO] " M "\n", ##__VA_ARGS__)
```

Comment:
There are a lot of issues around macros that I don't need to remind you about.

For now at a bare minimum follow the THIS_MACRO_SCARES_CHILDREN naming syntax (you had it before...I don't know why you changed it) and also pick a more specific name so that collisions are avoided.

A much later TODO is to revisit how you guys want to do error logging. You should probably have some common solution to this. If it is a function in a util that does what this macro is, great. If you want something more intense like Boost::log, then that's also ok (and what we use).